src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs

```diff
@@ -0,0 +1,188 @@
+    internal class AllSourcesFilterHandlerFactory : IScopeFilterFactory
+    {
+        private const string AllSourcesScopeFilter = "All Sources";
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public AllSourcesFilterHandlerFactory()
+        {
+        }
+
+        public IErrorListFilterHandler CreateFilter(IWpfTableControl tableControl)
+        {
+            return new AllSourcesFilterHandler();
+        }
+    }
+
+    internal class AllSourcesFilterHandler : RoslynFilterHandler
+    {
+        private const int AllSourcesFilterHandlerFilterId = 22;
+
+        public AllSourcesFilterHandler()
+            : base(AllSourcesFilterHandlerFilterId, ServicesVSResources.All_sources, ItemOrigin.IndexedInThirdParty)
+        {
+        }
+
+        protected override void OnActivated()
+        {
+            base.OnActivated();
+            // User just selected the broadest scope. See if we need to re-do the search
+        }
+    }
+
+    [Export(typeof(IScopeFilterFactory))]
+    [TableManagerIdentifier("FindAllReferences*")]
+    [TableManagerIdentifier("FindResults*")]
+    [DeferCreation(OptionName = RichNavOptions.RichNavAvailableOptionName)] // This factory will not be loaded unless this option is set to Boolean true
+    [Name(PredefinedScopeFilterNames.EntireRepositoryScopeFilter)]
+    [Order(Before = PredefinedScopeFilterNames.LoadedSolutionScopeFilter)]
+    internal class EntireRepositoryFilterHandlerFactory : IScopeFilterFactory
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public EntireRepositoryFilterHandlerFactory()
+        {
+        }
+
+        public IErrorListFilterHandler CreateFilter(IWpfTableControl tableControl)
+        {
+            return new EntireRepositoryFilterHandler();
+        }
+    }
+
+    internal class EntireRepositoryFilterHandler : RoslynFilterHandler
+    {
+        private const int EntireRepositoryFilterHandlerFilterId = 21;
+
+        public EntireRepositoryFilterHandler()
+            : base(EntireRepositoryFilterHandlerFilterId, ServicesVSResources.Entire_repository, ItemOrigin.IndexedInRepo)
+        {
+        }
+
+        protected override void OnActivated()
+        {
+            base.OnActivated();
+            // User just selected a broader scope. See if we need to re-do the search
+        }
+    }
+
+    internal class ItemOriginFilter : IEntryFilter
+    {
+        private readonly ItemOrigin _targetOrigin;
+
+        internal ItemOriginFilter(ItemOrigin targetOrigin)
+        {
+            _targetOrigin = targetOrigin;
+        }
+
+        public bool Match(ITableEntryHandle entry)
+        {
+            Requires.NotNull(entry, nameof(entry));
+
+            if (entry.TryGetValue(StandardTableKeyNames.ItemOrigin, out ItemOrigin entryOrigin))
+            {
+                return entryOrigin < _targetOrigin;
```

Comment:
shouldn't this be `<=` @AmadeusW ?

Reply:
yes! return `entryOrigin <= _targetOrigin`